lib/min_heap.rb

```diff
@@ -8,26 +8,36 @@ def initialize(key, value)
 end
 
 class MinHeap
-
   def initialize
     @store = []
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(log n), where n is the number of levels in the heap
+  # Space Complexity: O(1), because there are no extra variables that change with input size
```

Comment:
Since you have a recursive `heap_up` method, you have to consider the space complexity of that method.  So... this is actually O(log n).

The same applies for all your recursive methods.